src/bb8_backup.py

```diff
@@ -0,0 +1,43 @@
+from subprocess import run
+
+targets = ["orderly"]
+finished_setup = False
+bb8_dir = "../montagu-bb8/bb8"
+
+
+def requires_bb8_setup(func):
+    def decorated(*args, **kwargs):
+        global finished_setup
+        if not finished_setup:
+            setup()
+        return func(*args, **kwargs)
+
+    return decorated
+
+
+def setup():
+    global finished_setup
+    template = "- Configuring and installing bb8 backup service with these " \
+               "targets: {}"
+    print(template.format(targets))
+    args = ["./setup.sh", "../config.json"] + targets
```

Comment:
this path algebra here is confusing me - this is `../config.json` from the point of view of `../montagu-bb8/bb8` which ends up being `../montagu-bb8/config.json` I think.  Would it be easier (and allow more easy moving of modules at a later date if wanted) to run the path of the config through `os.path.abspath` and pass that through as the argument?

Reply:
Like this?